### ISSUE TITLE
SIGSEGV on `linkerd upgrade config`

### ISSUE THREAD

**Issue by codeman9** [linkerd/linkerd2#3094]
### What is the issue?

`linkerd upgrade config` produces a SIGSEGV error:

linkerd was installed like so: `linkerd install -l tram-linkerd --ha --linkerd-cni-enabled --proxy-uid 12102` and the linkerd config map looks like this:

```yaml
apiVersion: v1
data:
  global: |
    {"linkerdNamespace":"tram-linkerd","cniEnabled":true,"version":"stable-2.4.0","identityContext":{"trustDomain":"cluster.local","trustAnchorsPem":"-----BEGIN CERTIFICATE-----CERTISHERE-----END CERTIFICATE-----\n","issuanceLifetime":"86400s","clockSkewAllowance":"20s"},"autoInjectContext":null,"omitWebhookSideEffects":false}
  install: |
    {"uuid":"95049da3-8d83-4889-a3db-02359690906c","cliVersion":"stable-2.4.0","flags":[{"name":"ha","value":"true"},{"name":"linkerd-cni-enabled","value":"true"},{"name":"proxy-uid","value":"12102"}]}
  proxy: |
    {"proxyImage":{"imageName":"gcr.io/linkerd-io/proxy","pullPolicy":"IfNotPresent"},"proxyInitImage":{"imageName":"gcr.io/linkerd-io/proxy-init","pullPolicy":"IfNotPresent"},"controlPort":{"port":4190},"ignoreInboundPorts":[],"ignoreOutboundPorts":[],"inboundPort":{"port":4143},"adminPort":{"port":4191},"outboundPort":{"port":4140},"resource":{"requestCpu":"100m","requestMemory":"20Mi","limitCpu":"","limitMemory":""},"proxyUid":"12102","logLevel":{"level":"warn,linkerd2_proxy=info"},"disableExternalProfiles":true,"proxyVersion":"stable-2.4.0","proxyInitImageVersion":"v1.0.0"}
kind: ConfigMap
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","data":{"global":"{\"linkerdNamespace\":\"tram-linkerd\",\"cniEnabled\":true,\"version\":\"stable-2.4.0\",\"identityContext\":{\"trustDomain\":\"cluster.local\",\"trustAnchorsPem\":\"-----BEGIN CERTIFICATE-----CERTISHERE-----END CERTIFICATE-----\\n\",\"issuanceLifetime\":\"86400s\",\"clockSkewAllowance\":\"20s\"},\"autoInjectContext\":null,\"omitWebhookSideEffects\":false}\n","install":"{\"uuid\":\"95049da3-8d83-4889-a3db-02359690906c\",\"cliVersion\":\"stable-2.4.0\",\"flags\":[{\"name\":\"ha\",\"value\":\"true\"},{\"name\":\"linkerd-cni-enabled\",\"value\":\"true\"},{\"name\":\"proxy-uid\",\"value\":\"12102\"}]}\n","proxy":"{\"proxyImage\":{\"imageName\":\"gcr.io/linkerd-io/proxy\",\"pullPolicy\":\"IfNotPresent\"},\"proxyInitImage\":{\"imageName\":\"gcr.io/linkerd-io/proxy-init\",\"pullPolicy\":\"IfNotPresent\"},\"controlPort\":{\"port\":4190},\"ignoreInboundPorts\":[],\"ignoreOutboundPorts\":[],\"inboundPort\":{\"port\":4143},\"adminPort\":{\"port\":4191},\"outboundPort\":{\"port\":4140},\"resource\":{\"requestCpu\":\"100m\",\"requestMemory\":\"20Mi\",\"limitCpu\":\"\",\"limitMemory\":\"\"},\"proxyUid\":\"12102\",\"logLevel\":{\"level\":\"warn,linkerd2_proxy=info\"},\"disableExternalProfiles\":true,\"proxyVersion\":\"stable-2.4.0\",\"proxyInitImageVersion\":\"v1.0.0\"}\n"},"kind":"ConfigMap","metadata":{"annotations":{"linkerd.io/created-by":"linkerd/cli stable-2.4.0"},"labels":{"linkerd.io/control-plane-component":"controller","linkerd.io/control-plane-ns":"tram-linkerd"},"name":"linkerd-config","namespace":"tram-linkerd"}}
    linkerd.io/created-by: linkerd/cli stable-2.4.0
  creationTimestamp: "2019-07-17T16:13:12Z"
  labels:
    linkerd.io/control-plane-component: controller
    linkerd.io/control-plane-ns: tram-linkerd
  name: linkerd-config
  namespace: tram-linkerd
  resourceVersion: "4935"
  selfLink: /api/v1/namespaces/tram-linkerd/configmaps/linkerd-config
  uid: c6babc71-a8ad-11e9-93f1-0aeda684a428
```

### How can it be reproduced?

For me, just running `linkerd upgrade config` created the error. `linkerd upgrade control-plane` works fine.

### Logs, error output, etc

```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xe0 pc=0x1da5485]

goroutine 1 [running]:
github.com/spf13/pflag.(*FlagSet).GetNormalizeFunc(0x0, 0x10)
    /go/src/github.com/spf13/pflag/flag.go:217 +0x5
github.com/spf13/pflag.(*FlagSet).normalizeFlagName(0x0, 0xc00003d450, 0x2, 0xc00055cd80, 0x0)
    /go/src/github.com/spf13/pflag/flag.go:224 +0x2b
github.com/spf13/pflag.(*FlagSet).Lookup(0x0, 0xc00003d450, 0x2, 0x0)
    /go/src/github.com/spf13/pflag/flag.go:319 +0x3f
github.com/linkerd/linkerd2/cli/cmd.setFlagsFromInstall(0x0, 0xc0006b68c0, 0x3, 0x3)
    /go/src/github.com/linkerd/linkerd2/cli/cmd/upgrade.go:269 +0x73
github.com/linkerd/linkerd2/cli/cmd.(*upgradeOptions).validateAndBuild(0xc0004f72a0, 0x2233e33, 0x6, 0x243b0a0, 0xc0005532c0, 0x0, 0x0, 0x0, 0xc00050f650, 0xc000639c80)
    /go/src/github.com/linkerd/linkerd2/cli/cmd/upgrade.go:206 +0x1b4
github.com/linkerd/linkerd2/cli/cmd.upgradeRunE(0xc0004f72a0, 0x2233e33, 0x6, 0x0, 0x0, 0x0)
    /go/src/github.com/linkerd/linkerd2/cli/cmd/upgrade.go:160 +0x29c
github.com/linkerd/linkerd2/cli/cmd.newCmdUpgradeConfig.func1(0xc0005a5200, 0xc0004f7580, 0x0, 0x2, 0x0, 0x0)
    /go/src/github.com/linkerd/linkerd2/cli/cmd/upgrade.go:69 +0x48
github.com/spf13/cobra.(*Command).execute(0xc0005a5200, 0xc0004f7520, 0x2, 0x2, 0xc0005a5200, 0xc0004f7520)
    /go/src/github.com/spf13/cobra/command.go:698 +0x47a
github.com/spf13/cobra.(*Command).ExecuteC(0x305d180, 0xc00009c058, 0x0, 0xc00009c480)
    /go/src/github.com/spf13/cobra/command.go:783 +0x2dc
github.com/spf13/cobra.(*Command).Execute(0x305d180, 0xc000639f88, 0xc00009c058)
    /go/src/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
    /go/src/github.com/linkerd/linkerd2/cli/main.go:12 +0x2d
```

#### `linkerd check` output

```text
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
× control plane PodSecurityPolicies exist
    missing PodSecurityPolicies: linkerd-tram-linkerd-control-plane
    see https://linkerd.io/checks/#l5d-existence-psp for hints

Status check results are ×
```

The missing PSP is not an issue for us as we use our own PSP. The control plane works fine.

### Environment

- Kubernetes Version: 1.14.3
- Cluster Environment: (GKE, AKS, kops, ...): AWS
- Host OS: CoreOS 2079.6.0
- Linkerd version: stable-2.4.0

**Comment by alpeb** [comment]
Thanks @codeman9 for the report. I could reproduce the problem. Will report back with a possible solution.

**Comment by alpeb** [comment]
@codeman9 This has been fixed in edge-19.7.4, released today.
